### ISSUE TITLE
Cinnamenu@json - new web history search option can cause / partition to fill completely in some circumstances.

### ISSUE THREAD

**Issue by smurphos** [linuxmint/cinnamon-spices-applets#4287]
```
 * Applet version/Build date: Version 4.15.4 (2022-03-22 22:09:29)
 * Cinnamon version: 5.2.7
 * Distribution - Mint 20.3
 * Graphics:  Device-1: Intel UHD Graphics 620 driver: i915 v: kernel
           Device-2: Realtek Integrated_Webcam_HD type: USB driver: uvcvideo
           Display: x11 server: X.Org 1.20.13 driver: loaded: modesetting unloaded: fbdev,vesa
           resolution: 1920x1080~60Hz
           OpenGL: renderer: Mesa Intel UHD Graphics 620 (WHL GT2) v: 4.6 Mesa 21.2.6
 ```

**Notify author of applet**
@fredcw 

**Issue**

Here's a good one for you ;-)

I suspect this issue may be manifesting in the way it does because of my specific configuration so here are the those details.
8GB RAM machine.
/tmp is mounted in tmpfs (i.e held in RAM)
No physical swap - machine uses a ZRAM based swap (i.e it compresses memory rather than swapping it to disk).
I don't delete browsing history, so there is a lot of it - browser is google-chrome-stable.

So yesterday I noted sudden increases in the recorded SWAP usage (due to /tmp in RAM filling up) and also /var/tmp filling up (I guess it acts as an overflow when /tmp is full?) with corrupted very large (upto 3.7Gb) sqlite3 files with fairly random names. This can quickly get to the point that I'm warned that my / partition is full. On examination of the files I see they contain lots of text strings associated with my web browsing.

Using `inotifywait /var/tmp/ -m -e create` and `inotifywait /tmp/ -m -e create` to watch the directories I've narrowed the source down to the new web history search in Cinnamenu -  these files are created as the search box is typed into. Most of them are transient, but some (always with a filename starting tmp) persist and these can end up filling /tmp and overflowing into /var/tmp. I've just crashed my system putting together this bug report with half a dozen uses of the search box as the / partition filled.

![2022-03-28_07-16](https://user-images.githubusercontent.com/29017677/160338354-e3890122-2be4-44e7-b52c-34e6b0456e05.png)

**Steps to reproduce**
See above

**Expected behaviour**

**Other information**

As further troubleshooting I'll get Chrome installed in a Cinnamon VM with a  regular storage hosted /tmp and swap file , 'create' some browsing history and see what happens there.

**Comment by smurphos** [comment]
OK in the VM it just doesn't work - searches that would be expected to bring up some history are putting this message into .xsession-errors -

```
(cinnamon:1404): St-CRITICAL **: 08:12:30.367: cr_parser_new_from_buf: assertion 'a_buf && a_len' failed

(cinnamon:1404): St-CRITICAL **: 08:12:30.367: cr_declaration_parse_list_from_buf: assertion 'parser' failed

(cinnamon:1404): St-CRITICAL **: 08:12:30.368: cr_parser_new_from_buf: assertion 'a_buf && a_len' failed
```

**Comment by fredcw** [comment]
Ok, thanks. I have a solution to the problem of left behind tmp files. I don't think it's anything to do with your tmpfs setup, but rather just an error reading either your browser's History or Favicons file so I should have allowed for this exception anyway.

Not sure what's causing the problem, maybe it's just the large size of these files. My ~/.config/google-chrome/Default/History and ~/.config/google-chrome/Default/Favicons files are about 30MB and 25MB respectively, and this is with default google history settings I think (18 months). I found a google page that gives the "remember history" options of 3/18/36 months or never delete which I presume is the option you use. It would be interesting to know how big these files are in your case.

As for the VM problem, cinnamenu searches for the following files to get search history:

<user config dir>/chromium/Default/History
<user config dir>/google-chrome/Default/History
<user config dir>/opera/History
<user config dir>/vivaldi/Default/History
<user config dir>/BraveSoftware/Brave-Browser/Default/History
<user config dir>/microsoft-edge/Default/History

but checks first for the relevant desktop file, eg. usr/share/applications/google-chrome.desktop.

if these files exist then I don't know why it isn't finding your history in this case.

If there's an error reading the Favicons file then the history results should still show, just with a normal google-chrome icon instead.

**Comment by fredcw** [comment]
I've just submitted a PR which should fix the tmp files problem. #4289

**Comment by smurphos** [comment]
Thanks fredcw - my History file is 1.7GiB ;-) (history in Google Account was set to never delete)

Favicons is much slimmer 28MiB

On the VM - yep the bookmark search is also failing. google-chrome.desktop is present and correct as are the History and Favicon files in the profile. Will investigate further.

**Comment by fredcw** [comment]
@smurphos yes, bookmarks and history are working on Manjaro but not on Mint, I've found the problem.

**Comment by smurphos** [comment]
Both issues fixed in 4.15.5 (2022-04-08 20:07:50)

**Comment by smurphos** [comment]
Thanks @fredcw
